internal/observability/middleware.go

```diff
@@ -181,8 +181,12 @@ func applyLogLevelsConfig(dst *levels, src *DirectionalLevelsConfig) {
 
 // Handle implements middleware.UnaryInbound.
 func (m *Middleware) Handle(ctx context.Context, req *transport.Request, w transport.ResponseWriter, h transport.UnaryHandler) error {
-	call := m.graph.begin(ctx, transport.Unary, _directionInbound, req)
-	defer m.handlePanicForCall(call, transport.Unary)
+
+	var call call
+	m.graph.begin(ctx, transport.Unary, _directionInbound, req, &call)
+	// It is safe to pass a pointer to defer because the stack memory doesn't get released so there is no escape to the
+	// heap
+	defer m.handlePanicForCall(&call, transport.Unary)
```

Comment:
is it not easier to change `m.handlePanicForCall` to `c.handlePanicForCall` ?

it doesn't seem to depend on anything in `m`

Reply:
Not sure, I wanted to touch as little as possible the original code.